src/diff/component.js

```diff
@@ -27,173 +30,41 @@ export function renderComponent(
-
-	if (internal && internal._component) {
-		c = internal._component;
-	} else {
-		// Instantiate the new component
-		if ('prototype' in type && type.prototype.render) {
-			// @ts-ignore The check above verifies that newType is suppose to be constructed
-			internal._component = c = new type(newProps, componentContext); // eslint-disable-line new-cap
-		} else {
-			// @ts-ignore Trust me, Component implements the interface we want
-			internal._component = c = new Component(newProps, componentContext);
-			c.constructor = type;
-			c.render = doRender;
-		}
-		if (provider) provider.sub(c);
-
-		c.props = newProps;
-		if (!c.state) c.state = {};
-		c.context = componentContext;
-		c._globalContext = globalContext;
-		isNew = true;
-		internal._flags |= DIRTY_BIT;
-	}
-
-	// Invoke getDerivedStateFromProps
-	if (c._nextState == null) {
-		c._nextState = c.state;
-	}
-	if (type.getDerivedStateFromProps != null) {
-		if (c._nextState == c.state) {
-			c._nextState = assign({}, c._nextState);
-		}
-
-		assign(c._nextState, type.getDerivedStateFromProps(newProps, c._nextState));
-	}
-
-	oldProps = c.props;
-	oldState = c.state;
-	if (isNew) {
-		if (type.getDerivedStateFromProps == null && c.componentWillMount != null) {
-			c.componentWillMount();
-		}
-
-		if (c.componentDidMount != null) {
-			// If the component was constructed, queue up componentDidMount so the
-			// first time this internal commits (regardless of suspense or not) it
-			// will be called
-			addCommitCallback(internal, c.componentDidMount);
-		}
-	} else {
-		if (
-			type.getDerivedStateFromProps == null &&
-			newProps !== oldProps &&
-			c.componentWillReceiveProps != null
-		) {
-			c.componentWillReceiveProps(newProps, componentContext);
-		}
-
-		if (
-			(!(internal._flags & FORCE_UPDATE) &&
-				c.shouldComponentUpdate != null &&
-				c.shouldComponentUpdate(newProps, c._nextState, componentContext) ===
-					false) ||
-			(newVNode && newVNode._vnodeId === internal._vnodeId)
-		) {
-			c.props = newProps;
-			c.state = c._nextState;
-			internal.props = newProps;
-			// More info about this here: https://gist.github.com/JoviDeCroock/bec5f2ce93544d2e6070ef8e0036e4e8
-			if (newVNode && newVNode._vnodeId !== internal._vnodeId) {
-				internal._flags &= ~DIRTY_BIT;
-			}
-
-			c._internal = internal;
-			if (
-				internal._commitCallbacks != null &&
-				internal._commitCallbacks.length
-			) {
-				commitQueue.push(internal);
-			}
+	let prevContext = rendererState.context;
+	rendererState.skip = false;
+	rendererState.commit = false;
+	rendererState.force = (internal._flags & FORCE_UPDATE) == FORCE_UPDATE;
 
-			// TODO: Returning undefined here (i.e. return;) passes all tests. That seems
-			// like a bug. Should validate that we have test coverage for sCU that
-			// returns Fragments with multiple DOM children
-			return reorderChildren(internal, startDom, parentDom);
-		}
+	const renderResult = renderReactComponent(newVNode, internal, rendererState);
```

Comment:
I wonder if it would be worth differentiating and making a hot path: if class `renderComponent` if func `type.render()`